torch/_inductor/ir.py

```diff
@@ -4113,6 +4113,123 @@ def codegen(self, wrapper):
         wrapper.writeline(f"_register_tensor_work({output_name}, {output_name}_work)")
 
 
+class MultiOutputNoSizeAssert(MultiOutput):
+    """
+    Extract partial output from a multi-output OP.
+        Works like MultiOutput but doesn't assert size. This must be a property guaranteed by the op emiting this.
```

Comment:
curious why we don't need size asserts in collectives for the multi output cases?

Reply:
Because we know that the expected size matches the runtime one since we control the allocation.